src/device.rs

```diff
@@ -676,26 +719,57 @@ pub extern "C" fn wgpuDeviceCreateSwapChain(
     };
     let error = gfx_select!(device => GLOBAL.surface_configure(surface, device, &config));
     if let Some(error) = error {
-        panic!("Failed to configure surface: {}", error);
+        handle_device_error(device, &error);
+        None
+    } else {
+        SURFACE_TO_DEVICE.lock().unwrap().insert(surface, device);
+        Some(surface)
     }
-    SURFACE_TO_DEVICE.lock().unwrap().insert(surface, device);
-    surface // swap chain_id == surface_id
+}
+
+#[derive(Debug, Error)]
+pub enum SurfaceError {
+    #[error("Surface timed out")]
+    Timeout,
+    #[error("Surface is outdated")]
+    Outdated,
+    #[error("Surface was lost")]
+    Lost,
 }
 
 #[no_mangle]
 pub extern "C" fn wgpuSwapChainGetCurrentTextureView(
     swap_chain: id::SurfaceId,
 ) -> Option<id::TextureViewId> {
     let surface_id = swap_chain;
-    let device_id = get_device_from_surface(surface_id);
-    let result =
-        gfx_select!(device_id => GLOBAL.surface_get_current_texture(surface_id, PhantomData));
-    let texture = result
-        .expect("Unable to get swap chain texture view")
-        .texture_id
-        .unwrap();
-    let desc = wgc::resource::TextureViewDescriptor::default();
-    Some(gfx_select!(texture => GLOBAL.texture_create_view(texture, &desc, PhantomData)).0)
+    let device = get_device_from_surface(surface_id);
+    match gfx_select!(device => GLOBAL.surface_get_current_texture(surface_id, PhantomData)) {
+        Err(error) => {
+            handle_device_error(device, &error);
+            return None;
```

Comment:
why do we have all those returns? just end with an expression

Reply:
Apologies, I'm very new to Rust, but when I tried without `return`, I get a `expected enum 'Option', found '()'` compile error. Somehow it doesn't resolve them to `Option<id::TextureViewId>` without a return.

Reply:
You'd just leave `None` here without `return` or a semicolon